HW3

DedaLenya opened the pull request:
Урок 3. Урок 3. Вебинар. Введение в проектирование БД

Commit:
Практическое задание
Урок 3. Урок 3. Вебинар. Введение в проектирование БД

tigal commented on HW3/HW3_Task2.sql:
```diff
@@ -0,0 +1,202 @@
+	from_user_id bigint UNSIGNED NOT NULL ,
+	visible_all boolean DEFAULT TRUE,
+	txt_post text NOT NULL,
+	media_id int UNSIGNED DEFAULT NULL, 
```
Не хватает foreign key на медиа, раз даете ссылку)

tigal commented on HW3/HW3_Task2.sql:
```diff
@@ -0,0 +1,202 @@
+ 
+DESCRIBE new_post_user ;
+
+CREATE TABLE likes (	
```
Потеряли primary key, а он нам очень нужен:(
Таблицы для лайков на медиа и на посты должны быть отдельными. В вашем случае, как я понимаю, вы хотели сделать лайк на пост и на медиа в этому посту. Но что будет, если пользователь уберет лайк с медиа, а на посту его оставит ( у вас же в посту еще и текст будет).
Так что рекомендую от медиа пока отказаться в этой таблице.
Также, как я и сказала, потерян primary key. А нам нужно соблюсти условие уникальности, чтобы пользователь не мог поставить 10 лайков на 1 пост. Этого мы можем добиться именно через primary key, так, как это реализовано в comminities_users (можете туда подглядеть)

DedaLenya replied:
Сделал эти изменения, а ошибку эту так и не могу "поймать" что я делаю не так.

Error occurred during SQL script execution

Причина:
 SQL Error [3780] [HY000]: Referencing column 'media_id' and referenced column 'id' in foreign key constraint 'fk_media_id' are incompatible.

Commit:
Работа над ошибками по ревью

Commit:
Работа над ошибками по ревью2.
Ошибку нашёл, нужно было int заменить на bigint, надо будет понять в чём их различия.
2. Уникальность лайков поправил.

DedaLenya reviewed (commented):
Ошибку нашёл, нужно было int заменить на bigint, надо будет понять в чём их различия.
2. Уникальность лайков поправил.
